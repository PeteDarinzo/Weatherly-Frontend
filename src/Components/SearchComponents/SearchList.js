import React, { useEffect, useState } from "react";
import SearchCard from "./SearchCard";
import Container from '@mui/material/Container';
import Stack from '@mui/material/Stack';

import { makeStyles } from "@mui/styles";
import { Typography } from "@mui/material";

const useStyles = makeStyles({
  container: {
    padding: 20
  }
});


const SearchList = ({ movies, saveMovie }) => {

  const classes = useStyles();

  if (movies === undefined) {
    return (
      <Typography>No results</Typography>
    )
  }

  return (
    <Container className={classes.container} maxWidth="sm">
      <Stack>
        {movies.map(movie => (<SearchCard movie={movie} saveMovie={saveMovie} key={movie.imdbID} />))}
      </Stack>
    </Container>
  );
}

export default SearchList;