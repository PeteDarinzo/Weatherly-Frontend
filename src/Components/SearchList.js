import React, { useEffect, useState } from "react";
import SearchCard from "./SearchCard";
import Container from '@mui/material/Container';
import Stack from '@mui/material/Stack';

const SearchList = ({ movie, saveMovie }) => {

  return (
    <Container maxWidth="sm">
      <Stack spacing={2}>
        {Object.keys(movie).length ? (<SearchCard movie={movie} saveMovie={saveMovie} />) : (<b>Search for some movies!</b>)}
      </Stack>
    </Container>
  );
}

export default SearchList;