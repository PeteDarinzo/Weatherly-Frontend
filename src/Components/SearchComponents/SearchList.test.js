import SearchList from "./SearchList";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

it("renders a search list without crashing", () => {
  render(
    <MemoryRouter>
      <SearchList movies={[]} saveMovie={() => { }} />
    </MemoryRouter>
  );
});


it("matches the search list snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchList movies={[]} saveMovie={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
