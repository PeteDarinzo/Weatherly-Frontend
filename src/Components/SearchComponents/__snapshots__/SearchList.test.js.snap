// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the search list snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm makeStyles-container-2 css-cuefkz-MuiContainer-root"
  >
    <div
      class="css-nen11g-MuiStack-root"
    />
  </div>
</DocumentFragment>
`;
